azure-toolkit-libs/azure-toolkit-redis-lib/src/main/java/com/microsoft/azure/toolkit/redis/AzureRedis.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
+package com.microsoft.azure.toolkit.redis;
+
+import com.azure.core.management.exception.ManagementError;
+import com.azure.core.management.exception.ManagementException;
+import com.azure.resourcemanager.redis.RedisManager;
+import com.azure.resourcemanager.redis.fluent.RedisClient;
+import com.azure.resourcemanager.redis.models.CheckNameAvailabilityParameters;
+import com.microsoft.azure.toolkit.lib.AzureService;
+import com.microsoft.azure.toolkit.lib.SubscriptionScoped;
+import com.microsoft.azure.toolkit.lib.common.entity.CheckNameAvailabilityResultEntity;
+import com.microsoft.azure.toolkit.lib.common.event.AzureOperationEvent;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+import com.microsoft.azure.toolkit.lib.common.operation.AzureOperation;
+import com.microsoft.azure.toolkit.lib.common.task.ICommittable;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class AzureRedis extends SubscriptionScoped<AzureRedis> implements AzureService, AzureOperationEvent.Source<AzureRedis> {
+    public AzureRedis() {
+        super(AzureRedis::new);
+    }
+
+    private AzureRedis(@Nonnull final List<Subscription> subscriptions) {
+        super(AzureRedis::new, subscriptions);
+    }
+
+    @Nonnull
+    @AzureOperation(name = "redis.list.subscription|selected", type = AzureOperation.Type.SERVICE)
+    public List<RedisCache> list() {
+        return getSubscriptions().stream()
+            .map(subscription -> RedisManagerFactory.create(subscription.getId()))
+            .flatMap(manager -> manager.redisCaches().list().stream())
+            .map(redis -> new RedisCache(redis.manager(), redis))
+            .collect(Collectors.toList());
+    }
+
+    @AzureOperation(name = "redis.check_name", params = "name", type = AzureOperation.Type.SERVICE)
+    public CheckNameAvailabilityResultEntity checkNameAvailability(String subscriptionId, String name) {
+        final RedisManager redisManager = RedisManagerFactory.create(subscriptionId);
+        RedisClient redis = redisManager.redisCaches().manager().serviceClient().getRedis();
+        try {
+            redis.checkNameAvailability(new CheckNameAvailabilityParameters().withName(name).withType("Microsoft.Cache/redis"));
+            return new CheckNameAvailabilityResultEntity(true, null);
+        } catch (ManagementException ex) {
+            ManagementError value = ex.getValue();
+            if (value != null && "NameNotAvailable".equals(value.getCode())) {
```

Comment:
error code equals "NameNotAvailable"?

Reply:
It is caused by redis check name api

Reply:
it is defined by server side rest api, we have no way to change it